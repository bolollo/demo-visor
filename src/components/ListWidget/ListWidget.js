// @flow

import React from "react";
import PropTypes from "prop-types";

import styles from "./ListWidget.module.css";

export default function ListWidget({items, selected, position, handleClick, locale, handleSelectInvert, handleSelectNone, handleSelectAll}) {

	const positionClassName = styles[position];

	function isSelected (item) {

		return selected.find(sel => item.ogc_fid === sel.id);

	}

	return (
		<div className={[styles.listWidget, positionClassName].join(" ")}>
			<div className={styles.selector}>
				<span onClick={handleSelectAll}>ALL</span><span onClick={handleSelectNone}>NONE</span><span onClick={handleSelectInvert}>INVERT</span>
			</div>
			<ul>
				{items.map(item => (
					<li key={item.ogc_fid} onClick={handleClick.bind(this, item)} className={isSelected(item) ? styles.selected : null}>
						{item.nombre_igg} <span className={styles.small}>({item.p_total.toLocaleString(locale)})</span>
					</li>
				))}
			</ul>
		</div>
	);

}

ListWidget.propTypes = {
	items: PropTypes.array,
	selected: PropTypes.array,
	position: PropTypes.oneOf(["bottom-right", "bottom-left", "top-right", "top-left", "relative"]),
	handleClick: PropTypes.func,
	locale: PropTypes.string,
	handleSelectInvert: PropTypes.func,
	handleSelectNone: PropTypes.func,
	handleSelectAll: PropTypes.func
};

ListWidget.defaultProps = {
	items: [],
	selected: [],
	position: "top-left",
	handleClick: () => {},
	locale: "es-ES",
	handleSelectInvert: () => {},
	handleSelectNone: () => {},
	handleSelectAll: () => {}
};
