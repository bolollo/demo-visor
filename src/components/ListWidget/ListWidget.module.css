.listWidget {
    z-index:100;
    margin: 0.5rem;
    padding: 5px;
    background-color: var(--primary-white, #ffffff);
    border-radius: 5px;
    min-width: 150px;
    opacity: 0.9;
    height: calc(100% - 30px);
    overflow-y: auto;
    width: 170px;
}

.listWidget ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
    font-size: 0.85em;
}

.listWidget ul li{
    cursor: pointer;
}

.selector {
    margin-bottom: 5px;
    border-bottom: 1px solid;
    border-bottom-color: var(--light-grey, #6c757d);
    text-align: center;
}

.selector span{
    margin-right: 5px;
    color:var(--light-grey, #6c757d);
    cursor: pointer;
    font-size: 0.7em;
}

.selected{
    font-weight: bold;
}

.small {
    font-size: 0.6em;
    color: var(--light-grey, #6c757d);
}

.bottom-right{
    position: absolute;
    bottom: 0;
    right: 0;
}
.bottom-left{
    position: absolute;
    bottom: 0;
    left: 0;
}
.top-right{
    position: absolute;
    top: 0;
    right: 0;
}
.top-left{
    position: absolute;
    top: 0;
    left: 0;
}
.relative{
    position: relative;
}