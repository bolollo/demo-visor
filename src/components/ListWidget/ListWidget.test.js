// @flow

import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import ListWidget from "./ListWidget";

const items = [
	{"ogc_fid":22, "nombre_igg":"MÃ©xico", "p_total":15175862},
	{"ogc_fid":28, "nombre_igg":"Distrito Federal", "p_total":8851080},
	{"ogc_fid":7, "nombre_igg":"Veracruz", "p_total":7643194}
];

describe("ListWidget component", () => {

	it("matches the snapshot ", () => {

		const component = shallow(<ListWidget></ListWidget>);
		expect(toJSON(component)).toMatchSnapshot();

	});

	it("has count equal 3", () => {

		const component = shallow(<ListWidget items={items}></ListWidget>);
		expect(component.find("li")).toHaveLength(3);

	});

	it("has position bottom-left", () => {

		const component = shallow(<ListWidget position={"bottom-left"}></ListWidget>);
		expect(component.find(".listWidget").hasClass("bottom-left")).toEqual(true);

	});

	it("Test click li", () => {

		const mockClick = jest.fn();
		const component = shallow(<ListWidget items={items} handleClick={mockClick}></ListWidget>);
		component.find("li").at(1).simulate("click");
		expect(mockClick).toHaveBeenCalled();

	});

	it("Test click selectAll", () => {

		const mockClick = jest.fn();
		const component = shallow(<ListWidget handleSelectAll={mockClick}></ListWidget>);
		component.find(".selector span").at(0).simulate("click");
		expect(mockClick).toHaveBeenCalled();

	});

	it("Test click selectNone", () => {

		const mockClick = jest.fn();
		const component = shallow(<ListWidget handleSelectNone={mockClick}></ListWidget>);
		component.find(".selector span").at(1).simulate("click");
		expect(mockClick).toHaveBeenCalled();

	});

	it("Test click selectInvert", () => {

		const mockClick = jest.fn();
		const component = shallow(<ListWidget handleSelectInvert={mockClick}></ListWidget>);
		component.find(".selector span").at(2).simulate("click");
		expect(mockClick).toHaveBeenCalled();

	});

});
