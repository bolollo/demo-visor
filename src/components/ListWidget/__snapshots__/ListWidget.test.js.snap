// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListWidget component matches the snapshot  1`] = `
<div
  className="listWidget top-left"
>
  <div
    className="selector"
  >
    <span
      onClick={[Function]}
    >
      ALL
    </span>
    <span
      onClick={[Function]}
    >
      NONE
    </span>
    <span
      onClick={[Function]}
    >
      INVERT
    </span>
  </div>
  <ul />
</div>
`;
